File: example/python/lib/rest.py
import os
import requests
import lib.utils as utils

service_url = "http://localhost:"+str(os.getenv("NT_HTTP_PORT"))+"/api/v6"
headers = {
  "Content-Type": "application/json; charset=utf-8"
}

def decodeResult(resultStr, code):
  result = utils.checkJson(resultStr)
  if code not in [200, 201, 204]:
    return [None, result]
  return [result, None]

def set_headers(token):
  if token != "":
    headers["Authorization"] = "Bearer "+token
  else:
    headers["X-API-KEY"] = os.getenv("NT_API_KEY")

def Get(token, path, query):
  set_headers(token)
  try:
    path = service_url+"/"+path
    query_str = "&".join([f"{k}={v}" for k, v in query.items()])
    if query_str != "":
      path += "?"+query_str
    response = requests.get(path, headers=headers)
    return decodeResult(response.text, response.status_code)
  except requests.ConnectionError:
    return [None, "Connection failure"]

def Post(token, path, data):
  set_headers(token)
  try:
    path = service_url+"/"+path
    response = requests.post(path, headers=headers, json=data)
    return decodeResult(response.text, response.status_code)
  except requests.ConnectionError:
    return [None, "Connection failure"]

def Put(token, path, data):
  set_headers(token)
  try:
    path = service_url+"/"+path
    response = requests.put(path, headers=headers, json=data)
    return decodeResult(response.text, response.status_code)
  except requests.ConnectionError:
    return [None, "Connection failure"]

def Delete(token, path):
  set_headers(token)
  try:
    path = service_url+"/"+path
    response = requests.delete(path, headers=headers)
    return decodeResult(response.text, response.status_code)
  except requests.ConnectionError:
    return [None, "Connection failure"]

File: example/php/lib/rest.php
<?php

use Symfony\Component\HttpClient\HttpClient;

class Rest {
  function request($method, $path, $token, $query = array(), $data = array()){
    $service_url = 'http://localhost:'.getenv('NT_HTTP_PORT').'/api/v6/'.$path;
    $header = array(
      'Content-Type: application/json',
      'Accept: application/json'
    );
    if ($token != '') {
      $header = array_merge($header, array('Authorization: Bearer '.$token));
    }
    if ($token == '') {
      $header = array_merge($header, array('X-API-Key: '.getenv('NT_API_KEY')));
    }

    $client = HttpClient::create();
    $response = $client->request($method, $service_url, [
      'headers' => $header,
      'query' => $query,
      'body' => json_encode($data, JSON_UNESCAPED_UNICODE)
    ]);

    $status = $response->getStatusCode();
    if($status != 200 && $status != 201 && $status != 204) {
      return array(
        'result' => NULL,
        'error' => $response->getContent(false)
      );
    }
    $headers = $response->getHeaders();
    if(array_key_exists('content-type', $headers)) {
      $contentType = $headers['content-type'][0];
      if($contentType == 'application/pdf') {
        return array(
          'result' => $response->getContent(false),
          'error' => NULL
        );
      }
    }
    
    $body = $response->getContent();
    $response = json_decode($body, true);
    return array(
      'result' => $response,
      'error' => NULL
    );
  }

  function error($f3) {
    return array(
      'result' => NULL,
      'error' => $f3->get('ERROR.text')
    );
  }

  public function Get($token='', $path='', $query = array()){
    return $this->request('GET', $path, $token, $query);
  }

  public function Post($token='', $path='', $data = NULL){
    return $this->request('POST', $path, $token, array(), $data);
  }

  public function Put($token='', $path='', $data = NULL){
    return $this->request('PUT', $path, $token, array(), $data);
  }

  public function Delete($token='', $path=''){
    return $this->request('DELETE', $path, $token, array());
  }
}

?>